app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :email, format: URI::MailTo::EMAIL_REGEXP
+
+  # the authenticate method from devise documentation
+  def self.authenticate(email, password)
```

Comment:
Why do we need this method, if it provied by default from devise it should be automatically available?

Reply:
This is a part for Doorkeeper (on the API task). Not needed for Devise but used in Dookeerper config:
![image](https://user-images.githubusercontent.com/77609814/121848240-f8f7e380-cd13-11eb-89c1-8a6223c83cbc.png)

Reply:
I followed this tutorial: https://rubyyagi.com/rails-api-authentication-devise-doorkeeper/

Reply:
And the methods is not "in devise code" but it is provided by Devise as a way to do:
https://github.com/heartcombo/devise/wiki/How-To:-Find-a-user-when-you-have-their-credentials